fdb-record-layer-lucene/src/test/java/com/apple/foundationdb/record/lucene/LucenOnlineIndexingTest.java

```diff
@@ -0,0 +1,221 @@
+    void luceneOnlineIndexingTest4() {
+        luceneOnlineIndexingTestAny(TEXT_AND_STORED, COMPLEX_DOC, 8);
+    }
+
+    private String randomText(Random rn) {
+        switch (rn.nextInt() % 4) {
+            case 0: return ENGINEER_JOKE;
+            case 1: return WAYLON;
+            case 2: return WAYLON + " says who?";
+            case 3: return ENGINEER_JOKE + " is it really funny?";
+            default: return ENGINEER_JOKE + WAYLON ;
+        }
+    }
+
+    private int randomGroup(Random rn) {
+        return rn.nextInt() % 2;
+    }
+
+
+    void luceneOnlineIndexingTestAny(Index index, String document, int numRecords) {
+        assertTrue(numRecords > 3);
+        final Random rn = new Random();
+        rn.nextInt();
+        disableIndex(index, document);
+
+        long[] docIds = new long[numRecords * 2];
+        for (int i = 0; i < numRecords * 2; i ++) {
+            docIds[i] = rn.nextLong();
+        }
+        int middle = 1 + rn.nextInt(numRecords - 2);
+        // write records
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            for (int i = 0; i < numRecords; i ++) {
+                long docId = docIds[i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            context.commit();
+        }
+        // overwrite some records (to enforce merge), write others as new
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            for (int i = 0; i < middle; i ++) {
+                long docId = docIds[i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            for (int i = middle; i < numRecords; i ++) {
+                long docId = docIds[numRecords + i];
+                recordStore.saveRecord(createSimpleDocument(docId, randomText(rn), randomGroup(rn)));
+            }
+            context.commit();
+        }
+        // build the index ..
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            try (OnlineIndexer indexBuilder = OnlineIndexer.newBuilder()
+                    .setRecordStore(recordStore)
+                    .setIndex(index)
+                    .setIndexingPolicy(OnlineIndexer.IndexingPolicy.newBuilder()
+                            .setPreventLuceneMergeDuringIndexing(true)
+                            .build())
+                    .build()) {
+                assertTrue(recordStore.isIndexDisabled(index));
+                indexBuilder.buildIndex(true);
+            }
+        }
+        // .. and assert readable mode
+        try (final FDBRecordContext context = openContext()) {
+            rebuildIndexMetaData(context, document, index);
+            assertTrue(recordStore.isIndexReadable(index));
```

Comment:
This also seems like it should assert that merge happens.

Reply:
I might be missing something - we do not allow merge while indexing in this test.

Reply:
This test calls `.setPreventLuceneMergeDuringIndexing(true)`
IIUC that means:
1. that it won't merge on close during the normal indexing
2. it will, after each transaction call merge, if one is needed (which I think it always will be)

So at the end, it should have done some merges.

I think if you add a check on the number of files, and run it a few times with (2) commented out you will see a very high segment count. If you then run it with that code uncommented, you should see a much lower number.
Repeating both runs a few times should give you an indication of the predictability of the file count, so that you can add an assertion.

Reply:
✅